diff --git a/tools/cmake/FindR.cmake b/tools/cmake/FindR.cmake
index df5c33e..c15e34b 100644
--- a/tools/cmake/FindR.cmake
+++ b/tools/cmake/FindR.cmake
@@ -35,7 +35,7 @@ set(TEMP_CMAKE_FIND_APPBUNDLE ${CMAKE_FIND_APPBUNDLE})
 set(CMAKE_FIND_APPBUNDLE "NEVER")

 # Find R.
-find_program(R_EXECUTABLE R DOC "R executable.")
+set(R_EXECUTABLE $ENV{R})

 if(R_EXECUTABLE)
     # Get the location of R.
--- a/configure
+++ b/configure
@@ -32,7 +32,7 @@ CXXFLAGS="$CPPFLAGS $CXXFLAGS $EXTRACXXFLAGS -D_LIBCPP_DISABLE_AVAILABILITY"
 #LDFLAGS="$R_LDFLAGS $R_SHLDFLAGS"

 cd src
-$CMAKE ../tools \
+$CMAKE --debug-find ../tools \
   -DCMAKE_BUILD_TYPE=RelWithDebInfo \
   -DCMAKE_C_FLAGS_INIT="" \
   -DCMAKE_CXX_FLAGS_INIT="" \
