diff --git a/configure b/configure
index 4afa666..9adc5c3 100755
--- a/configure
+++ b/configure
@@ -43,4 +43,3 @@ $CMAKE ../tools \
   -DR_SHLIB_EXT="$R_SHLIB_EXT" \
   -DPASSTHROUGH_CONFIGURE_ARGS="$*" \
   ${CMAKE_ARGS}
-echo "" >> _deps/highfive-src/include/highfive/bits/H5Friends.hpp
diff --git a/configure.ucrt b/configure.ucrt
index 7b43c83..19f275f 100644
--- a/configure.ucrt
+++ b/configure.ucrt
@@ -28,5 +28,3 @@ cmake ../tools \
   -DR_HOME="$R_HOME"
 
 mv Makefile Makefile.ucrt
-
-echo "" >> _deps/highfive-src/include/highfive/bits/H5Friends.hpp
