diff --git a/configure.ucrt b/configure.ucrt
index 7b43c83..fe7ab75 100644
--- a/configure.ucrt
+++ b/configure.ucrt
@@ -13,5 +13,5 @@
 LDFLAGS=`"${R_HOME}/bin/R" CMD config LDFLAGS`
 FLIBS=`"${R_HOME}/bin/R" CMD config FLIBS`
-
+LDFLAGS="$LDFLAGS -fopenmp"
 CFLAGS="$CPPFLAGS $CFLAGS"
 CXXFLAGS="$CPPFLAGS $CXXFLAGS"
@@ -22,7 +22,7 @@
 cd src
-cmake ../tools \
+cmake -G "MinGW Makefiles" ../tools -DBLA_VENDOR="Generic" \
   -DCMAKE_BUILD_TYPE=RelWithDebInfo \
-  -DCMAKE_C_FLAGS_INIT="" \
-  -DCMAKE_CXX_FLAGS_INIT="" \
+  -DCMAKE_C_FLAGS="-I${PREFIX}/Library/include" \
+  -DCMAKE_CXX_FLAGS="-I${PREFIX}/Library/include" \
   -DCMAKE_POSITION_INDEPENDENT_CODE:bool=ON \
   -DR_HOME="$R_HOME"
