diff --git a/configure.ucrt b/configure.ucrt
index 7b43c83..fe7ab75 100644
--- a/configure.ucrt
+++ b/configure.ucrt
@@ -19,7 +19,7 @@ RTOOLS=`"${R_HOME}/bin/R" CMD config R_TOOLS_SOFT`
 FC="$RTOOLS"/bin/gfortran.exe

 cd src
-cmake ../tools \
+cmake -G "MinGW Makefiles" ../tools -DBLA_VENDOR="Generic" \
   -DCMAKE_BUILD_TYPE=RelWithDebInfo \
   -DCMAKE_C_FLAGS_INIT="" \
   -DCMAKE_CXX_FLAGS_INIT="" \
